feat(ado-ext-telemetry): add a v1 to v2 migration guide for the ADO extension (#1136)
microsoft/accessibility-insights-action: docs/ado-extension-usage.md

If you want to run the extension multiple times in a single pipeline, you will need to ensure that unique `artifactName` and `outputDir` inputs are specified in your YAML file for each task. Artifact names and the output directory must be unique across all tasks in a pipeline.

## Migrating from version 1 to version 2

Version 2.x of the extension contains several breaking changes from version 1.x. To migrate, you will need to make a few adjustments based on whether your pipeline is defined using a YAML file or the "Classic" Pipelines web interface:

### Migrating a YAML Pipeline definition

> 1. The `repoServiceConnectionName` input has been removed. If you previously created a Service Connection specfically for this task, you should delete it under your Azure DevOps Project's "Service Connections" settings.

Reviewer: small typo
```suggestion
1. The `repoServiceConnectionName` input has been removed. If you previously created a Service Connection specifically for this task, you should delete it under your Azure DevOps Project's "Service Connections" settings.
```

> 1. The `repoServiceConnectionName` input has been removed. If you previously created a Service Connection specfically for this task, you should delete it under your Azure DevOps Project's "Service Connections" settings.

Reviewer: Some users may have granted the default build service ADO account repo:write permissions instead of creating a service connection. I'm not sure if we should mention that explicitly, that usage pattern should be removed as well.

> 2. The task inputs related to specifying a "static" site to scan (`siteDir`, `port`, and `urlRelativePath`) have changed to make it more clear that they are related (and mutually exclusive with `url`).

Reviewer: git blame shows it may have been `scanUrlRelativePath` and `localhostPort`?, [seen here](https://github.com/microsoft/accessibility-insights-action/blame/61e5b2ace73e3563405ea6876b1c1b64f5759843/packages/ado-extension/task.json#L88)
 
```suggestion
2. The task inputs related to specifying a "static" site to scan (`siteDir`, `localhostPort`, and `scanUrlRelativePath`) have changed to make it more clear that they are related (and mutually exclusive with `url`).
```

    - If you previously used a separate `publish` step to upload the `_accessibility-reports` folder, you can delete that `publish` step
    - If your pipeline is running in OneBranch, or any other environment where individual tasks cannot publish artifacts directly, specify `uploadResultsAsArtifact: false` to skip the new automatic artifact uploading
    - See [Report Artifacts](#report-artifacts) for more details, including how to customize the artifact name
4. By default, the task now fails if it detects an accessibility failure (unless the failure is a known issue tracked by a [Baseline File](#using-a-baseline-file))
    - If you previously specified `failOnAccessibilityError: true`, you can remove it (this is now the default behavior)
    - If you would prefer to keep the old behavior, where accessibility issues are not treated as a task failure, you can add `failOnAccessibilityError: false` (but consider [using a Baseline File](#using-a-baseline-file) instead!)

### Migrating a "Classic" Pipeline definition

> 1. The "Azure Repos Connection" option has been removed. If you previously created a Service Connection specfically for this task, you should delete it under your Azure DevOps Project's "Service Connections" settings.

Reviewer: small typo
```suggestion
1. The "Azure Repos Connection" option has been removed. If you previously created a Service Connection specifically for this task, you should delete it under your Azure DevOps Project's "Service Connections" settings.
```

2. The options related to specifying which site to scan have moved underneath a new "Hosting Mode" option to make it more clear which ones can be used together.
    - If you previously specified a "Site Directory", select the `staticSite` "Hosting Mode"
        - The "Site Directory", "Localhost Port" and "Scan URL Relative Path" task inputs now appear only when `staticSite` is selected
    - If you previously specified a "Website URL", select the `dynamicSite` "Hosting Mode"
        - The "Website URL" option now appears only when `dynamicSite` is selected

>     - If you previously specified _both_ as "Site Directory" and a "Website URL", you had a misconfiguration - these options were mutually exclusive, and the "Website URL" options was being silently ignored. Select `staticSite` mode and ignore your old "Website URL" input

Reviewer: (nit) "options was" might sound better as "options were" or "option was"
```suggestion
    - If you previously specified _both_ as "Site Directory" and a "Website URL", you had a misconfiguration - these options were mutually exclusive, and the "Website URL" option was being silently ignored. Select `staticSite` mode and ignore your old "Website URL" input
```
